src/item.cpp

```diff
@@ -1076,34 +1078,82 @@ std::string item::info(bool showtext, std::vector<iteminfo> &dump_ref) const
         it_tool* tool = dynamic_cast<it_tool*>(type);
 
         if ((tool->max_charges)!=0) {
-            std::string charges_line = _("Charges"); //;
+            int t_max;
```

Comment:
Both variants has strengths and weaknesses. I like first one, but this is not obvious behaviour. Second is obvious, but require many lines in every logic block...

Guess, i will choose moving call of dump->push_back() to end of logic. This is not so strange.